mne/viz/backends/renderer.py

```diff
@@ -292,6 +293,20 @@ def get_brain_class():
     """
     if get_3d_backend() == "mayavi":
         from surfer import Brain
+        _require_version('surfer', 'stc.plot', '0.9')
+        # Here we patch to avoid segfault:
+        # https://github.com/mne-tools/mne-python/pull/8828
+        close_func = Brain.close
+
+        def patched_close(self, *args, **kwargs):
+            try:
+                vtk_widget = self._figures[0][0].scene._content
+                with _qt_disable_paint(vtk_widget):
+                    close_func(self, *args, **kwargs)
+            except (AttributeError, IndexError):
+                close_func(self, *args, **kwargs)
+
+        Brain.close = patched_close
```

Comment:
This patches it at the class level which is not great. I would do it after instantiating so you only patch instances created by MNE

Reply:
We have examples using `get_brain_class()`. Those won't have the patched version.

Reply:
I think that's okay. This is really a patch to get tests to be reliable. We are mostly abandoning use of PySurfer anyway. I'd rather mess with it as little as possible.